profile/attribute/image_slice_attribute.go

```diff
@@ -42,18 +46,23 @@ func NewImageSlice(a *yotiprotoattr.Attribute) (*ImageSliceAttribute, error) {
 }
 
 // CreateImageSlice takes a slice of Items, and converts them into a slice of images
-func CreateImageSlice(items []*Item) (result []*media.Image) {
+func CreateImageSlice(items []*Item) (result []media.Media, err error) {
 	for _, item := range items {
 
-		imageValue := item.GetValue().(*media.Image)
-
-		result = append(result, imageValue)
+		switch i := item.Value.(type) {
+		case media.PNGImage:
+			result = append(result, i)
+		case media.JPEGImage:
+			result = append(result, i)
+		default:
+			return nil, fmt.Errorf("unexpected item type %T", i)
+		}
```

Comment:
Even if we don't go with this approach, I think it's definitely good to have this error handling here anyway 👍